templates/servicedefaults-crd.yaml

```diff
@@ -0,0 +1,137 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    cert-manager.io/inject-ca-from: {{ .Release.Namepace }}/{{ template "consul.fullname" . }}-controller
+    controller-gen.kubebuilder.io/version: v0.2.5
+  creationTimestamp: null
+  name: servicedefaults.consul.hashicorp.com
+spec:
+  additionalPrinterColumns:
+    - JSONPath: .status.conditions[?(@.type=="Synced")].status
+      description: The sync status of the resource with Consul
+      name: Synced
+      type: string
+  conversion:
+    strategy: None
+  group: consul.hashicorp.com
+  names:
+    kind: ServiceDefaults
+    listKind: ServiceDefaultsList
+    plural: servicedefaults
+    singular: servicedefaults
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      description: ServiceDefaults is the Schema for the servicedefaults API
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          description: ServiceDefaultsSpec defines the desired state of ServiceDefaults
+          properties:
+            expose:
+              description: ExposeConfig describes HTTP paths to expose through Envoy
+                outside of Connect. Users can expose individual paths and/or all HTTP/GRPC
+                paths for checks.
+              properties:
+                checks:
+                  description: Checks defines whether paths associated with Consul
+                    checks will be exposed. This flag triggers exposing all HTTP and
+                    GRPC check paths registered for the service.
+                  type: boolean
+                paths:
+                  description: Paths is the list of paths exposed through the proxy.
+                  items:
+                    properties:
+                      listenerPort:
+                        description: ListenerPort defines the port of the proxy's
```

Comment:
Yes it's taken from the golang structs which are capitalized. I think best to leave it as is because that is proper godoc, what do you think?
```
type ExposePath struct {
	// ListenerPort defines the port of the proxy's listener for exposed paths.
	ListenerPort int `json:"listenerPort,omitempty"`
```

Reply:
Yeah, in that case, let's leave it as is.